File: crafting_with_solid_principles/code/python/3-l.py

class Rectangle:
    def __init__(self, width, height):
        self.width = 0
        self.height = 0

    def set_dimension(self, width, height):
        self.width = width
        self.height = height

    def area(self):
        return self.width * self.height


class Square(Rectangle):
  def set_dimension(self, width, height):
      if width != height:
          raise "Invalid argument"
      super(self).set_dimension(width, width)

# ---

def enlarge(rectangle):
    height = rectangle.height;
    rectangle.set_dimension(height * 3, height)

File: crafting_with_solid_principles/code/java/3-l.java
public class Rectangle {
  private int width, height;
  public void setDimension(int width, int height) {
    this.width = width;
    this.height = height;
  }
  public int area() {
    return width * height;
  }
}

public class Square extends Rectangle {
  @Override
  public void setDimension(int width, int height) {
    if (width != height)
      throw new IllegalArgumentException();
    super.setDimension(width, width);
  }
}

// ---

public int enlarge(Rectangle rectangle) {
  int height = rectangle.height;
  rectangle.setDimension(height * 3, height)
}
